wandb/sdk/launch/sweeps/scheduler_sweep.py

```diff
@@ -120,40 +119,55 @@ def _heartbeat(self, worker_id: int) -> None:
                 else:
                     self.state = SchedulerState.FAILED
                     raise SchedulerError(f"AgentHeartbeat unknown command type {_type}")
+        return True
 
     def _run(self) -> None:
         # Go through all workers and heartbeat
-        for worker_id in self._workers.keys():
-            self._heartbeat(worker_id)
-        try:
-            run: SweepRun = self._heartbeat_queue.get(
-                timeout=self._heartbeat_queue_timeout
-            )
-        except queue.Empty:
-            wandb.termlog(f"{LOG_PREFIX}No jobs in Sweeps RunQueue, waiting...")
-            time.sleep(self._heartbeat_queue_sleep)
-            return
-        # If run is already stopped just ignore the request
-        if run.state in [
-            SimpleRunState.DEAD,
-            SimpleRunState.UNKNOWN,
-        ]:
-            return
-        wandb.termlog(
-            f"{LOG_PREFIX}Converting Sweep Run (RunID:{run.id}) to Launch Job"
-        )
-        _ = self._add_to_launch_queue(
-            run_id=run.id,
-            entry_point=["python3", run.program] if run.program else None,
-            # Use legacy sweep utilities to extract args dict from agent heartbeat run.args
-            config={
-                "overrides": {
-                    "run_config": LegacySweepAgent._create_command_args(
-                        {"args": run.args}
-                    )["args_dict"]
-                }
-            },
-        )
+        for worker_id in self._workers:
+            if not self._heartbeat(worker_id):
+                break
+
+        for _worker_id in self._workers:
+            try:
+                run: SweepRun = self._heartbeat_queue.get(
+                    timeout=self._heartbeat_queue_timeout
+                )
+
+                # If run is already stopped just ignore the request
+                if run.state in [
+                    SimpleRunState.DEAD,
+                    SimpleRunState.UNKNOWN,
+                ]:
+                    if self.state != SchedulerState.FLUSH_RUNS:
```

Comment:
Can we put a comment here explaining why we're only going to attempt to launch this run if we're in the FLUSH_RUNS state?